rpm/__init__.py

```diff
@@ -79,21 +93,77 @@ def try_path(path: str) -> bool:
     return False
 
 
+def import_helper(path: Path, suffixes: List) -> None:
+    """
+    Helps import rpm in cases where the shared object file extensions include the python
+    interpreter version
+
+    Args:
+        path (Path): the absolute path to the rpm module
+        suffixes (List): a list of extension suffixes valid for the python interpreter
+
+    Returns:
+        None
+    """
+    try:
+        importlib.reload(sys.modules[__name__])
+    except ModuleNotFoundError as e:
+        if e.name is None:
+            raise
+        logger.debug(
+            f"Failed to import {e.name} from {path}, "
+            "looking for extensions with valid suffixes"
+        )
+        stem = path / e.name.split(".")[1]
+        for suffix in suffixes:
+            so = Path(str(stem) + suffix)
+            logger.debug(f"Looking for {so}")
+            if not so.exists():
+                continue
+            load_module_by_path(e.name, so)
+            importlib.reload(sys.modules[__name__])
+            return
+        else:
+            logger.debug(f"Give up on {path}")
+
+
+def load_module_by_path(module_name: str, path: Path) -> None:
+    """
+    Imports a python module by file path
+
+    Args:
+        module_name (str): name of the module to import
+        path (Path): absolute path to the module to import
+
+    Returns:
+        None
+    """
+    logger.debug(f"importing {path} as {module_name}")
+    spec = importlib.util.spec_from_file_location(module_name, path)
+    if spec is None:
+        return
+    if spec.loader is None:
+        return
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
```

Comment:
```suggestion
    module = importlib.util.module_from_spec(spec)
    spec.loader.exec_module(module)
    sys.modules[__name__] = module
```

The module is loaded by `exec_module()`, it just needs to be referenced from `sys.modules` in place of the original module (the shim). There is no need to call `importlib.reload()` after this.